src/webgpu/api/operation/resource_init/buffer.spec.ts

```diff
@@ -717,11 +730,200 @@ GPUBuffer, all the contents in that GPUBuffer have been initialized to 0.`
     renderPass.endPass();
     t.queue.submit([encoder.finish()]);
 
-    t.expectSingleColor(outputTexture, 'rgba8unorm', {
+    t.CheckBufferAndOutputTexture(indexBuffer, bufferSize, outputTexture);
+  });
+
+g.test('indirect_buffer_for_draw_indirect')
+  .desc(
+    `Verify when we use a GPUBuffer as an indirect buffer for drawIndirect() just after the creation
+of that GPUBuffer, all the contents in that GPUBuffer have been initialized to 0.`
+  )
+  .paramsSubcasesOnly(u => u.combine('bufferOffset', [0, 16]))
+  .fn(async t => {
+    const { bufferOffset } = t.params;
+
+    const renderPipeline = t.CreateRenderPipelineForTest(
+      t.device.createShaderModule({
+        code: `
+    struct VertexOut {
+      [[location(0)]] color : vec4<f32>;
+      [[builtin(position)]] position : vec4<f32>;
+    };
+
+    [[stage(vertex)]] fn main() -> VertexOut {
+      var output : VertexOut;
+      output.color = vec4<f32>(1.0, 0.0, 0.0, 1.0);
+      output.position = vec4<f32>(0.0, 0.0, 0.0, 1.0);
+      return output;
+    }`,
+      }),
+      false
+    );
+
+    const kDrawIndirectParametersSize = 16;
+    const bufferSize = kDrawIndirectParametersSize + bufferOffset;
+    const indirectBuffer = t.device.createBuffer({
+      size: bufferSize,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.INDIRECT,
+    });
+
+    const outputTexture = t.device.createTexture({
+      format: 'rgba8unorm',
       size: [1, 1, 1],
-      exp: { R: 0.0, G: 1.0, B: 0.0, A: 1.0 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
     });
 
-    const expectedBufferData = new Uint8Array(bufferSize);
-    t.expectGPUBufferValuesEqual(indexBuffer, expectedBufferData);
+    // Initialize outputTexture to green.
+    const encoder = t.device.createCommandEncoder();
+    t.RecordInitializeTextureColor(encoder, outputTexture, { r: 0.0, g: 1.0, b: 0.0, a: 1.0 });
+
+    const renderPass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: outputTexture.createView(),
+          loadValue: 'load',
+          storeOp: 'store',
+        },
+      ],
+    });
+    renderPass.setPipeline(renderPipeline);
+    renderPass.drawIndirect(indirectBuffer, bufferOffset);
+    renderPass.endPass();
+    t.queue.submit([encoder.finish()]);
+
+    // The indirect buffer should be lazily cleared to 0, so we actually draw nothing and the color
+    // attachment will keep its original color (green) after we end the render pass.
+    t.CheckBufferAndOutputTexture(indirectBuffer, bufferSize, outputTexture);
+  });
+
+g.test('indirect_buffer_for_draw_indexed_indirect')
```

Comment:
This can probably be folded inside `indirect_buffer_for_draw_indirect` with an `indexed` boolean. There's a ton of similarities.